src/extension.ts

```diff
@@ -88,8 +89,35 @@ class CatCodingPanel {
         this._extensionUri = extensionUri;
 
         // Set the webview's initial html content
+        this._document = vscode.window.activeTextEditor.document;
         this._update();
 
+        // Update the webview based on text changes
+        const changeDocumentSubscription = vscode.workspace.onDidChangeTextDocument(e => {
+            if (e.document.uri.toString() === this._document.uri.toString()) {
+                this._update();
+            }
+        });
+
+        // Make sure we get rid of the listener when our editor is closed.
+        // TODO Should we be using the extra arguments like in the subscription below?
+        // Or push this into _disposables instead?
+        // Do the multiple calls to onDidDispose stack or overwrite the listeners?
+        this._panel.onDidDispose(() => {
+            changeDocumentSubscription.dispose();
+        });
+
+        const switchDocumentSubscription = vscode.window.onDidChangeActiveTextEditor(e => {
+            if (e.document == undefined) {
+                this._document = e.document;
```

Comment:
Am I missing something or this line is executed in both cases?

Reply:
Yeah, I think this was a remnant of trying to handle the `undefined` case differently, but it can go outside the `if`. If we change the `_update` to empty the contents if the document is undefined, then we can even merge the two branches.